src/orion/algo/skopt/bayes.py

```diff
@@ -1,52 +1,152 @@
+        if not shape:
+            shape = (1,)
+        # Unpack dimension
+        for i in range(shape[0]):
+            dimensions.append(
+                dimension_class(
+                    name=key + '_' + str(i),
+                    prior=dimension._prior_name,
+                    low=low, high=high))
 
     return Space(dimensions)
 
 
+def pack_point(point, orion_space):
+    """Take a list of points and pack it appropriately as a point from `orion_space`.
+
+    :param point: array-like or list of numbers
+    :param orion_space: problem's parameter definition
+    """
+    packed = []
+    idx = 0
+    for dim in orion_space.values():
+        shape = dim.shape
+        if shape:
+            next_idx = idx + shape[0]
+            packed.append(tuple(point[idx:next_idx]))
+            idx = next_idx
+        else:
+            packed.append(point[idx])
+            idx += 1
+    return packed
+
+
+def unpack_point(point, orion_space):
+    """Take a list of points and pack it appropriately as a point from `orion_space`.
+
+    :param point: array-like or list of numbers
+    :param orion_space: problem's parameter definition
+    """
+    unpacked = []
+    for subpoint, dim in zip(point, orion_space.values()):
+        shape = dim.shape
+        if shape:
+            unpacked.extend(subpoint)
+        else:
+            unpacked.append(subpoint)
+    return unpacked
+
+
 class BayesianOptimizer(BaseAlgorithm):
     """Wrapper skopt's bayesian optimizer"""
 
-    def __init__(self, space, **kwargs):
-        super(BayesianOptimizer, self).__init__(space)
+    requires = 'real'
+
+    def __init__(self, space,
+                 strategy='cl_min', n_initial_points=10, acq_func="gp_hedge",
+                 **gp_kwargs):
+        """Initialize skopt's BayesianOptimizer.
+
+        Copying documentation from `skopt.learning.gaussian_process.gpr` and
+        `skopt.optimizer.optimizer`.
+
+        Parameters
+        ----------
+        space : `orion.algo.space.Space`
+           Problem's definition
+        strategy : str (default: cl_min)
+           Method to use to sample multiple points.
+           Supported options are `"cl_min"`, `"cl_mean"` or `"cl_max"`.
+           Check skopt docs for details.
+        n_initial_points : int (default: 10)
+           Number of evaluations of `func` with initialization points
+           before approximating it with `base_estimator`. Points provided as
+           `x0` count as initialization points. If len(x0) < n_initial_points
+           additional points are sampled at random.
+        acq_func : str (default: gp_hedge)
+           Function to minimize over the posterior distribution. Can be:
+           ``["LCB", "EI", "PI", "gp_hedge", "EIps", "PIps"]``. Check skopt
+           docs for details.
+        gp_kwargs : dict
+           Optional arguments passed to `skopt.learning.GaussianProcessRegressor`.
+
+        gp_kwargs
+        ---------
```

Comment:
Any idea how this would turn out in sphinx doc?

Reply:
As a sphinx header I believe, but it will not have any semantics for `sphinx.ext.numpydoc`.